pkg/ha/state/lease.go

```diff
@@ -69,44 +75,117 @@ func NewLease(c client.LeaseClient, cluster, potentialLeader string, minT, maxT,
 	}, nil
 }
 
-// UpdateLease uses the supplied client to attempt to update the lease for the potentialLeader.
+func (l *Lease) ValidateSamplesInfo(replica string, minT, maxT, currT time.Time) (bool, time.Time, error) {
+	l._mu.RLock()
+	currentLeader := l.leader
+	currentDeadline := l.leaseUntil
+	l._mu.RUnlock()
+
+	if currentLeader == replica {
+		if !maxT.Before(currentDeadline) {
+			if err := l.updateLease(replica, minT, maxT); err != nil {
+				return false, time.Time{}, err
+			}
+		}
+		return l.confirmLeaseValidation(replica, minT)
+	}
+
+	if !maxT.After(currentDeadline) {
+		return false, time.Time{}, nil
+	}
+
+	if err := l.TryChangeLeader(currT); err != nil {
+		return false, time.Time{}, err
+	}
+	return l.confirmLeaseValidation(replica, minT)
+}
+
+func (l *Lease) confirmLeaseValidation(replica string, minT time.Time) (bool, time.Time, error) {
+	l._mu.RLock()
+	defer l._mu.RUnlock()
```

Comment:
Nit: I like following this convention throughout this code
```suggestion
	l._mu.RLock()
	leader := l.leader
	leaseStart := l.leaseStart
	l._mu.RUnlock()
```